lib/api/core/statistics.js

```diff
@@ -145,23 +146,41 @@ module.exports = function (kuzzle) {
   };
 
   /**
-   * Gets all the saved statistics
+   * Gets the lasts saved statistics frame from a date
    *
    * @param requestObject
-   * @returns {Promise}
+   * @returns {Promise} promise resolved as a ResponseObject
    */
-  this.getAllStats = function (requestObject) {
+  this.getStats = function(requestObject) {
     var
       deferred = q.defer(),
       response = {
         statistics: {}
       },
-      frames;
-
+      frames,
+      currentDate,
+      startTime,
+      stopTime;
+
+    currentDate = new Date().getTime();
+    if (requestObject.data.body.startTime !== undefined) {
+      startTime = new Date(requestObject.data.body.startTime).getTime();
+    }
+    if (requestObject.data.body.stopTime !== undefined) {
+      stopTime = new Date(requestObject.data.body.stopTime).getTime();
+    }
+    if ((startTime !== undefined && isNaN(startTime)) || (stopTime !== undefined && isNaN(stopTime))) {
+      deferred.reject(new ResponseObject(requestObject, new InternalError('Invalid time value')));
+      return deferred.promise;
+    }
+    if (startTime >= currentDate) {
+      deferred.resolve(new ResponseObject(requestObject, response));
+      return deferred.promise;
+    }
     if (!this.lastFrame) {
-      // We return at least the current statistics frame
-      response.statistics[(new Date()).toISOString()] = this.currentStats;
-
+      if (!stopTime || stopTime >= currentDate) {
```

Comment:
I guess we should here append the currentStats, and not get back only this...
So remove the deffered.resolve() just behind ?